LL-7758 Improvements to the Header and SideBar

gsoares-ledger opened the pull request:
In order to integrate the rebranding of the Navigation sidebar and header ([LL-7758]) into Ledger Live, a few adaptations had to take place into the library:
* Make content props of the Header optional
* Add an optional notification badge in the sidebar items

[LL-7758]: https://ledgerhq.atlassian.net/browse/LL-7758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

qd-qd commented on packages/react/src/components/navigation/sideBar/Item/Item.tsx:
```diff
@@ -54,6 +54,14 @@ const ItemWrapper = styled.li`
   }
 `;
 
+const Badge = styled.div`
```
I've got the feeling that we can merge this one with the Badge component located here (packages/react/src/components/message/Notification/Badge.tsx). By rewriting it a bit and making the icon as an optional props for example.

WDYT?

gsoares-ledger replied:
I thought about that, but it's such a simple component that I'm not entirely sure it's worth it. 
That one is a bit more complex, with a different size too, I feel that adapting it to be usable here would make it even more complex.

Do you feel there's a strong reason to merge them?

qd-qd replied:
I thought we used this tiny component anywhere else in the app but I just double checked and I was wrong. So I agree with you, this would be overkill to merge them if we only need this variant here

jelbaz-ledger commented on packages/react/src/components/navigation/sideBar/Item/Item.tsx:
```diff
@@ -95,6 +111,7 @@ const Item = ({ label, children, onClick, isActive, isDisabled }: ItemType): JSX
       >
         <ItemLabel variant="paragraph">{label}</ItemLabel>
       </TransitionInOut>
+      {displayNotificationBadge && <Badge />}
```
Wondering if the badge should not be inside the transition scope 🤔

gsoares-ledger replied:
I does if we want it to disappear with the Text. I actually did it now, in order to fix the positioning when it's collapsed, I make the one beside the text disappear and another one appear in the right corner of the icon. It's the exact same behaviour we have in the current Ledger Live menu.

jelbaz-ledger reviewed (changes requested):
When the sidebar is closed I think that it looks a bit weird.

<img width="144" alt="Capture d’écran 2021-11-10 à 11 37 37" src="https://user-images.githubusercontent.com/86958797/141098460-be12db3a-e9c4-4995-a59b-6e2263fb39d0.png">

Also, I think that it might be nice to give the ability to the library users to pass whatever JSX they want instead of just a forcing a `<Badge />`. But this is subjective so you can ignore this if you think it's uneeded 😉.

gsoares-ledger commented:
> When the sidebar is closed I think that it looks a bit weird.
> 
> <img alt="Capture d’écran 2021-11-10 à 11 37 37" width="144" src="https://user-images.githubusercontent.com/86958797/141098460-be12db3a-e9c4-4995-a59b-6e2263fb39d0.png">
> 
> Also, I think that it might be nice to give the ability to the library users to pass whatever JSX they want instead of just a forcing a `<Badge />`. But this is subjective so you can ignore this if you think it's uneeded 😉.

I've updated the positioning of the bad when it's closed so it's in the top right corner of the Icon. It's the same behaviour as today in Ledger Live. I also added a custom Badge prop for users of the library.

Here's what it looks like now:
![new-badge](https://user-images.githubusercontent.com/91940736/141151598-32dc4fb6-12ad-4ed4-8e9f-1d9639d59a20.gif)

jelbaz-ledger reviewed (approved):
Looking great! 👍
